pkg/serviceaccount/legacy.go

```diff
@@ -144,12 +144,21 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
 
+		// Track legacy secret-based long-lived service account tokens
+		autoGenerated := false
+		// Check if it's a auto-generated secret-based token
 		for _, ref := range serviceAccount.Secrets {
 			if ref.Name == secret.Name {
+				autoGenerated = true
 				warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
 				break
 			}
 		}
+		// It's a manually-created secret-based token
+		if !autoGenerated {
+			warning.AddWarning(ctx, "", "Use time-bound tokens from the TokenRequest API instead of manually created secret-based long lived tokens if possible.")
```

Comment:
I'm not sure I agree with warning here... this means someone is explicitly using a manually created token... we shouldn't add warnings unless we're 100% sure the caller can take action to resolve the warning, and they may not be able to update the caller to use refreshable/time-bound tokens

Reply:
We have discovered it is beneficial to categorize tokens into different types: automated generated, manually created, and obtained through TokenRequest APIs. Instead of adding a warning in the response and audit log, would it make sense to add extra logging for these different token types to the apiserver log? Thanks!

Reply:
We have discovered it is beneficial to categorize tokens into different types: automated generated, manually created, and obtained through TokenRequest APIs. Instead of adding a warning in the response and audit log, would it make sense to add extra logging for these different token types to the apiserver log? Thanks!

Reply:
In practice, we noticed quite a few (old) external systems, e.g., a CI/CD pipeline, store and use secret-based tokens to authenticate with Kubernetes. With such a warning, we hope the user could call `TokenRequest` API to fetch time bound tokens instead.

Reply:
I'd be more open to decorating the audit log for use of a legacy token than returning a user-facing warning

Reply:
> I'd be more open to decorating the audit log for use of a legacy token than returning a user-facing warning

Thank, Jordan. Could you provide your recommendation on how to enhance the audit log for manually created tokens? I'd happy to update the PR.

Reply:
you can follow the example of the information we add when using a stale time-bound token: https://github.com/kubernetes/kubernetes/blob/master/pkg/serviceaccount/claims.go#L208

Reply:
Hi Jordan, I reverted the changes to` legacy.go`. The updated PR tracks manually created tokens and adds extra info. to the audit log in `claim.go` instead as you suggested. Hopefully I did it right. Can you take a look? Thanks!

I'm not sure about how to test the change to the audit log. I haven't found a test for state tokens yet.

Reply:
sorry, I wasn't suggesting to add it to claims.go, I was linking to that as an example of how to add audit annotations

Reply:
> sorry, I wasn't suggesting to add it to claims.go, I was linking to that as an example of how to add audit annotations

Sorry for misunderstanding it.  How about adding it to `legacy.go`  after the warning for auto-generated tokens? We can  add audit annotations for both auto-generated and manually-created tokens?  Thanks.

Reply:
Hi @liggitt, I've updated `legacy.go` to add audit annotations and metrics for auto-generated and manually created tokens. Can you please take a look when you get a chance? Thanks!

Reply:
@liggitt , I've made the suggested changes. Could you review it when you have a chance? Thanks!

Reply:
Hi Jordan, any feedback would be greatly appreciated. Thanks!

Reply:
Thanks, Jordan! I've committed the suggested changes. Can you take another look please?